source/dhtslib/vcf/package.d

```diff
@@ -3,4 +3,139 @@ module dhtslib.vcf;
+/// enum HeaderTypesStrings = __traits(allMembers, HeaderTypes);
+///
+/// works but includes "None" which is throwing off indexing
+/// enum HeaderTypesStrings = [__traits(allMembers, HeaderTypes)];
+enum HeaderTypesStrings = [__traits(allMembers, HeaderTypes)][1..$];
+
+
+/// Represents the classification of a headerline
+///
+/// ##INFO=<Number=A, Type=Integer>
+///               =
+///
+/// if FIXED
+/// ##INFO=<Number=2, Type=Integer>
+///               =
+/// 
+/// Replacement for htslib BCF_VL_*
+enum HeaderLengths
+{
+    None =              -1,
+    Fixed =             BCF_VL_FIXED, /// variable length: fixed length
+    Variable =          BCF_VL_VAR, /// variable length: variable
+    OnePerAltAllele =   BCF_VL_A, /// variable length: one field per alt allele
+    OnePerGenotype =    BCF_VL_G, /// variable length: one field per genotype
+    OnePerAllele =      BCF_VL_R, /// variable length: one field per allele including ref
+}
+
+/// Strings for HDR_LENGTH
+static immutable  HeaderLengthsStrings = ["FIXED",".","A","G","R"];
+
+/// Used to index into bcf_hdr_t's id field of type bcf_idpair_t*[3]
+///
+/// i.e as used from VCFRecord where this.vcfheader is a VCFHeader:
+/// this.vcfheader.hdr.id[HeaderDictTypes.Id]
+///
+/// Replacement for htslib BCF_DT_*
+enum HeaderDictTypes
+{
+    Id =        BCF_DT_ID, /// dictionary type: ID
+    Contig =    BCF_DT_CTG, /// dictionary type: Contig
+    Sample =    BCF_DT_SAMPLE, /// dictionary type: SAMPLE
+}
+
+/// Used by InfoField (bcf_info_t) and FormatField (bcf_fmt_t) 
+/// to identify the underlying htslib/bcf1_t info and format data
+/// type and size. This data is stored in ubyte arrays.
+///
+/// 
+/// Replacement for htslib BCF_BT_*
+enum RecordType
+{
+    Null =   0,  /// null
+    Int8 =   BCF_BT_INT8,  /// int8
+    Int16 =  BCF_BT_INT16,  /// int16
+    Int32 =  BCF_BT_INT32,  /// int32
+    Int64 =  BCF_BT_INT64,  /// Unofficial, for internal use only per htslib headers 
+    Float =  BCF_BT_FLOAT,  /// float (32?)
+    Char =   BCF_BT_CHAR  /// char (8 bit)
+}
+
+/// Byte sizes for RecordType
+static immutable RecordTypeSizes = [0, byte.sizeof, short.sizeof, int.sizeof, long.sizeof, float.sizeof, 0, char.sizeof];
+alias RecordTypeToDType = AliasSeq!(null, byte, short, int, long, float, null, string);
+
+/// Replacement for htslib VCF_*
+enum VariantType
+{
+    Ref =           VCF_REF,  /// ref (e.g. in a gVCF)
+    Snp =           VCF_SNP,  /// SNP 
+    Mnp =           VCF_MNP,  /// MNP
+    Indel =         VCF_INDEL,  /// INDEL
+    Other =         VCF_OTHER,  /// other (e.g. SV)
+    Breakend =      VCF_BND, /// breakend
+    Overlap =       VCF_OVERLAP, /// overlapping deletion, ALT=* 
+}
+
+/// Levels identifiers for unpacking the underlying variable length
+/// data in the bcf1_t. Values are inclusive 
+/// i.e UnpackLevels.AltAllele unpacks all data before and including the ALT allele
+/// Replacement for htslib BCF_UN_*
+enum UnpackLevels

```

Comment:
Enum types are typically named in the singular (look at your others, `XXXType`). This makes them look more sensibile when typing function parameters

Reply:
addressed in 
99f6949